cocos/renderer/frame-graph/CallbackPass.h

```diff
@@ -0,0 +1,72 @@
+/****************************************************************************
+ Copyright (c) 2021 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#pragma once
+
+#include "DevicePassResourceTable.h"
+
+namespace cc {
+namespace framegraph {
+
+class Executable {
+public:
+    Executable() noexcept          = default;
+    virtual ~Executable()          = default;
+    Executable(Executable const &) = delete;
+    Executable(Executable &&)      = delete;
+    Executable &operator=(Executable const &) = delete;
+    Executable &operator=(Executable &&) = delete;
+
+    virtual void execute(DevicePassResourceTable const &resourceTable) noexcept = 0;
+};
+
+template <typename Data, typename ExecuteMethod>
+class CallbackPass final : public Executable {
+public:
+    explicit CallbackPass(ExecuteMethod &&execute) noexcept;
+    virtual void       execute(DevicePassResourceTable const &resourceTable) noexcept override;
+    inline Data const &getData() const noexcept { return _data; }
+    inline Data &      getData() noexcept { return _data; }
+
+private:
+    Data          _data;
+    ExecuteMethod _execute;
+};
+
+//////////////////////////////////////////////////////////////////////////
+
+template <typename Data, typename ExecuteMethod>
+CallbackPass<Data, ExecuteMethod>::CallbackPass(ExecuteMethod &&execute) noexcept
```

Comment:
execute should be a functional, right? If so, then it is better to guard it using `std::is_function`.

Reply:
Oh, i thought it does.